src/main/java/com/ebivariation/contigalias/service/AssemblyService.java

```diff
@@ -18,27 +18,90 @@
 
 import com.ebivariation.contigalias.dao.AssemblyDao;
 import com.ebivariation.contigalias.entities.AssemblyEntity;
+import com.ebivariation.contigalias.entities.ChromosomeEntity;
+import com.ebivariation.contigalias.repo.AssemblyRepository;
+import org.hibernate.LazyInitializationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 
 import java.io.IOException;
+import java.util.List;
 import java.util.Optional;
 
 @Service
 public class AssemblyService {
 
-    private final AssemblyDao assemblyDao;
+    private final AssemblyRepository repository;
+
+    private final AssemblyDao dao;
+
+    private final Logger logger = LoggerFactory.getLogger(AssemblyService.class);
 
     @Autowired
-    public AssemblyService(@Qualifier("ftpDao") AssemblyDao assemblyDao) {
-        this.assemblyDao = assemblyDao;
+    public AssemblyService(AssemblyRepository repository, AssemblyDao dao) {
```

Comment:
I feels weird having a service, a DAO and a repository; in the sense that what the DAO class does, maybe is not exactly a DAO, and DAOs and repositories are basically the same. However, asking for the DAO interface or however we name it (that has the FTP implementation and might have other future implementations) is a great usage of polymorphism, aligned with what I said about the open-close principle. Good design here :+1:. I know it feels trivial but this is the kind of things that improve the overall stability of the system.

Reply:
Fixed in commit 62c1ba388571bf66a2e938cf3d36ae7decbcfbc1